tempto-core/src/main/java/com/teradata/tempto/fulfillment/table/TableDefinitionsRepository.java

```diff
@@ -99,14 +101,73 @@ public TableDefinitionsRepository(Collection<TableDefinition> tableDefinitions)
 
     public <T extends TableDefinition> T register(T tableDefinition)
     {
-        checkState(!tableDefinitions.containsKey(tableDefinition.getName()), "duplicated table definition: %s", tableDefinition.getName());
-        tableDefinitions.put(tableDefinition.getName(), tableDefinition);
+        TableDefinitionRepositoryKey repositoryKey = asRepositoryKey(tableDefinition.getTableHandle());
+        checkState(!tableDefinitions.containsKey(repositoryKey), "duplicated table definition: %s", repositoryKey);
+        tableDefinitions.put(repositoryKey, tableDefinition);
         return tableDefinition;
     }
 
-    public TableDefinition get(String name)
+    public TableDefinition get(TableHandle tableHandle)
```

Comment:
Lets say you have a nation defined in repository with no schema. Then you want to create nation in schema `dupablada`. In that case you ask for table `dupablada.nation` and you get `nation` without schema, but table fulfiller will create that table in `dupablada` schema.

But when table is defined in some schema, then it can be created only in that schema. See `tpcds`. 

It is not ideal solution, but this what I achieved without introducing backward incompatibilities. The root problem is that TPCH and TPCDS contains `customer` table and I could not store two different tables with the same name in table repository. That is why customer for tpcds is defined in that schema, but customer for tpch is defined without a schema. When ask for:
 - `customer` you will definition from TPCH
 - `dupablada.customer` you will definition from TPCH
 - `tpcds.customer` you will definition from TPCDS